hs_collection_resource/tests/test_collection.py

```diff
@@ -0,0 +1,472 @@
+            owner=self.user1,
+            title='My Collection with missing required metadata'
+        )
+
+        self.resGen1 = create_resource(
+            resource_type='GenericResource',
+            owner=self.user1,
+            title='Gen 1'
+        )
+
+        self.resGen2 = create_resource(
+            resource_type='GenericResource',
+            owner=self.user1,
+            title='Gen 2'
+        )
+
+        self.resGen3 = create_resource(
+            resource_type='GenericResource',
+            owner=self.user1,
+            title='Gen 3'
+        )
+
+        self.resTimeSeries = create_resource(
+            resource_type='TimeSeriesResource',
+            owner=self.user1,
+            title='Test Time Series Resource'
+        )
+
+        self.resNetCDF = create_resource(
+                    resource_type='NetcdfResource',
+                    owner=self.user1,
+                    title='Test NetCDF Resource'
+                )
+
+        self.resGeoFeature = create_resource(
+                    resource_type='GeographicFeatureResource',
+                    owner=self.user1,
+                    title='Test Geographic Feature (shapefiles)'
+                )
+
+        self.resModelInstance = create_resource(
+                    resource_type='ModelInstanceResource',
+                    owner=self.user1,
+                    title='Test Model Instance Resource')
+
+        self.user2 = create_account(
+            'byu2@byu.edu',
+            username='user2',
+            password='mypassword2',
+            first_name='myfirstname2',
+            last_name='mylastname2',
+            superuser=False,
+            groups=[self.group]
+        )
+
+        self.resGen4 = create_resource(
+            resource_type='GenericResource',
+            owner=self.user2,
+            title='Gen 4'
+        )
+
+        base_url = "/hsapi/_internal/{0}"
+        self.url_to_update_collection = base_url + "/update-collection/"
+        self.url_to_collection_member_permission = base_url + "/collection-member-permission/{1}/"
+        self.url_to_set_resource_flag = base_url + "/set-resource-flag/"
+        self.url_to_delete_resource = base_url + "/delete-resource/"
+        self.url_to_update_collection_for_deleted_resources = base_url + "/update-collection-for-deleted-resources/"
+
+    def tearDown(self):
+        super(TestCollection, self).tearDown()
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
+
+    def test_collection_basic_functions(self):
+        # test basic collection class with different res types
+
+        self.assertEqual(self.resCollection.resources.count(), 0)
+        # add res to collection.resources
+        self.resCollection.resources.add(self.resGen1)
+        self.resCollection.resources.add(self.resGeoFeature)
+        # add collection to res.collections
+        self.resModelInstance.collections.add(self.resCollection)
```

Comment:
@zhiyuli Ignore my last part of comment above - I see collections property many-to-many relations defined on BaseResource model now.